app/routes/datamart/land.py

```diff
@@ -66,85 +73,62 @@ async def tree_cover_loss_by_driver_get(
-            )
-        else:
-            return ORJSONResponse(
-                status_code=200,
-                content={
-                    "data": {
-                        "self": f"/v0/land/tree-cover-loss-by-driver/{resource_id}",
-                        "treeCoverLossByDriver": {
-                            "Permanent agriculture": 10,
-                            "Hard commodities": 12,
-                            "Shifting cultivation": 7,
-                            "Forest management": 93.4,
-                            "Wildfires": 42,
-                            "Settlements and infrastructure": 13.562,
-                            "Other natural disturbances": 6,
-                        },
-                        "metadata": {
-                            "sources": [
-                                {"dataset": "umd_tree_cover_loss", "version": "v1.11"},
-                                {
-                                    "dataset": "wri_google_tree_cover_loss_by_drivers",
-                                    "version": "v1.11",
-                                },
-                                {
-                                    "dataset": "umd_tree_cover_density_2000",
-                                    "version": "v1.11",
-                                },
-                            ]
-                        },
-                    },
-                    "status": "success",
-                },
-            )
-    except FileNotFoundError:
-        return ORJSONResponse(
-            status_code=404,
-            content={
-                "status": "failed",
-                "message": "Not Found",
-            },
-        )
+    resource = await _get_resource(resource_id)
+
+    headers = {}
+    if resource["status"] == "pending":
+        headers = {"Retry-After": "1"}
+
+    return ORJSONResponse(
+        status_code=200,
+        headers=headers,
+        content={"data": resource, "status": "success"},
+    )
 
 
 @router.post(
     "/tree-cover-loss-by-driver",
     response_class=ORJSONResponse,
     response_model=Response,
     tags=["Land"],
-    deprecated=True,
 )
 async def tree_cover_loss_by_driver_post(
     data: TreeCoverLossByDriverIn,
+    background_tasks: BackgroundTasks,
     api_key: APIKey = Depends(get_api_key),
 ):
     """Create new tree cover loss by drivers resource for a given geostore and
     canopy cover."""
 
+    # check geostore is valid
+    try:
+        await get_geostore(data.geostore_id, GeostoreOrigin.rw)
+    except HTTPException:
+        raise HTTPException(
+            status_code=422,
+            detail=f"Geostore {data.geostore_id} can't be found or is not valid.",
+        )
+
     # create initial Job item as pending
     # trigger background task to create item
     # return 202 accepted
-    resource_id = _get_resource_id(data.geostore_id, data.canopy_cover)
+    dataset_version = DEFAULT_LAND_DATASET_VERSIONS | data.dataset_version
```

Comment:
At first I thought you were using a default OR the one provided, and thought you had it the wrong way round. But this is actually merging two dictionaries, right?

Reply:
Yes! New syntax in Python 3.9+ I just learned about, I think I got the ordering right based on my test results... basically merges two dicts, with the one on the right overwriting any clashing keys